cachelib/allocator/memory/CompressedPtr.h

```diff
@@ -89,8 +91,11 @@ class CACHELIB_PACKED_ATTR CompressedPtr {
   PtrType ptr_{kNull};
 
   // create a compressed pointer for a valid memory allocation.
-  CompressedPtr(uint32_t slabIdx, uint32_t allocIdx)
-      : ptr_(compress(slabIdx, allocIdx)) {}
+  CompressedPtr(uint32_t slabIdx,
+                uint32_t allocIdx,
+                bool isMultiTiered,
```

Comment:
`isMultiTiered` looks redundant. We are storing the tid at the MSB of the compressed ptr. In this case, we can just interpret `isMultiTiered = false` if `tid == 0`. Additional logic in `CompressedPtr` can be simplified a lot in this way.

Reply:
The bit-packing format changes for multi-tier compressed pointer. The 32nd bit is reserved for tid only for multi-tiered compressed pointer. When config is single-tiered, compressed pointer bit packing used the original design - 32nd bit is not reserved for tid.
tid ==0 doesn't mean cachelib is not multi-tiered.